azurerm/internal/services/kusto/kusto_iothub_data_connection_resource.go

```diff
@@ -133,6 +171,9 @@ func resourceKustoIotHubDataConnectionCreate(d *pluginsdk.ResourceData, meta int
 		IotHubConnectionProperties: &kusto.IotHubConnectionProperties{
 			IotHubResourceID:       utils.String(d.Get("iothub_id").(string)),
 			ConsumerGroup:          utils.String(d.Get("consumer_group").(string)),
+			TableName:              utils.String(d.Get("table_name").(string)),
+			MappingRuleName:        utils.String(d.Get("mapping_rule_name").(string)),
+			DataFormat:             kusto.IotHubDataFormat(d.Get("data_format").(string)),
```

Comment:
If these are not specified, should they be sent as blank strings or omitted from the request?

Reply:
Here's another resource implementing the same functionality, could potentially do exactly the same :)
https://github.com/terraform-providers/terraform-provider-azurerm/blob/3c98880fcc47305d2a5cc5cdf8d77d983cb35da2/azurerm/internal/services/kusto/kusto_eventhub_data_connection_resource.go#L263

Reply:
Example for the tests can be ~~stolen~~ borrowed from the same resource :)

Reply:
Great point, thanks!

I've already been cheating off the eventhub 😃

Reply:
Better well stolen than poorly invented they say ;)

Reply:
I've omitted the optional parameters if not set now and added the tests